docs/jsonplatform.md

```diff
@@ -0,0 +1,130 @@
+JSON platform {#jsonplat}
+=============
+
+JSON platform loading allows libmraa users who work with **simple** and unsupported
+boards to create JSON configuration files which libmraa understands, allowing
+libmraa to just 'work' with the unsupported platform. JSON platform **does not work**
+with boards which require advanced functions which overwrite default functionality or
+boards that require the mux functionality.
+
+Syntax
+------
+
+The JSON file that libmraa reads needs specific values to be set by default, they
+are required in every JSON file intended to be used to load the platform.
+The keywords are case sensative so be aware when writing the JSON file.
+Each heading is a top level key in the JSON file, each keys type is Array
+
+### platform
+|Key		|Type	|Required	|Description					|
+|---------------|-------|---------------|-----------------------------------------------|
+|id 		|int 	|no		| An ID that can be assigned			|
+|name		|string	|yes		| The platform name				|
+|pin_count	|int	|yes		| The total pin count				|
+|gpio_count	|int	|no		| The subset of pincount which able to do GPIO	|
+|i2c_count	|int	|no		| The number of usable I2C ports		|
+|aio_count	|int	|no		| The number of usable AIO ports		|
+|spi_count	|int	|no		| The number of usable SPI ports		|
+|uart_count	|int	|no		| The total UART ports				|
+|pwmDefPeriod	|int	|no		| The default PWM period			|
+|pwmMaxPeriod	|int	|no		| The max PWM period				|
+|pwmMinPeriod	|int	|no		| The min PWM period				|
+
+
+### layout
+
+**THIS INDEX'S THE PIN ARRAY**
+There needs to be a label for each available pin, e.g if pin_count=28 then there should be 28
+pin labels. Each JSON element under layout is index from the beginning this also means because
+of the one-to-one relationship between using a pin that we need to set pin 0 as an invalid pin.
```

Comment:
That statement is IMHO a bit overloaded and ambiguous because of that, Maybe rephrase?